docs/latest/design/extending-envoy-gateway.md

```diff
@@ -0,0 +1,176 @@
+    backendRefs:
+    - name: backend
+      port: 3000
+```
+
+In order to enable the translation of new resources introduced by extensions, Envoy Gateway provides hook points within the translation pipeline, where it calls out to extension services registered in the [EnvoyGateway config][] 
+if they specify an `apiGroup` that matches the `group` of an `ExtensionRef` filter. The extension will then be able to modify the xDS that Envoy Gateway generated and send back the 
+modified configuration. If there are no registered extensions that specify support for the `group` of an `ExtensionRef` filter then Envoy Gateway will treat it as an unknown resource 
+and provide an error to the user.
+
+**Note:** Currently (as of [v1beta1][]) Gateway API does not provide a means to specify the namespace or version of an object referenced as an `ExtensionRef`. The extension mechanism will assume that
+the namespace of any `ExtensionRef` is the same as the namespace of the `HTTPRoute` or `GRPCRoute` it is attached to rather than treating the `name` field of an `ExtensionRef` as a `name.namespace` string.
+If Gateway API adds support for these fields then the design of the Envoy Gateway extensions will be updated to support them.
+
+## Control Plane Extensions
+
+To provide certain extensions such as a new authentication filter it is also necessary to be able to extend the Control plane
+to deploy additional services that are required such as a new [external authorization][] server or rate limiting server.
+
+To power these use-cases, Envoy Gateway provides additional hook entrypoints to the Infra Manager which is responsible for dynamically creating the
+instances of Envoy Proxy. These hook points empower extensions to modify the Envoy Proxy deployment and service dynamically.
+
+In the example above where an extension developer wants to create a new `ExtensionRef` filter that provides authentication, they would likely deploy their own authorization service into the cluster
+and make use of the extension hooks to configure Envoy to use that service to authenticate requests. By also using extension points in the control plane, the extension can dynamically
+inject the authorization service as a sidecar to the Envoy deployments created by Envoy Gateway rather than deploy the authorization service in the Kubernetes cluster separately. Doing so would help to minimize the 
+latency between Envoy and the authentication service and also allows the end user to spend less time worrying about how things are deployed in the cluster.
+
+![Control Plane Extension Example](../images/extension-auth-example.png)
+
+## Known Challenges
+
+Extending Envoy Gateway by using an external extension server which makes use of hook points in Envoy Gateway does come with some know issues that will have to be worked around.
+
+- Extension server handling it's own state such as in the scenario where an extension establishes watches its own CRDs in the cluster can lead to a scenario where
```

Comment:
Is it possible to let Envoy Gateway wrap the custom CRD resources in the context when calling the extenstion server hooks?